bundles/org.openhab.binding.velux/src/main/java/org/openhab/binding/velux/internal/handler/VeluxHandler.java

```diff
@@ -125,4 +129,37 @@ public void handleConfigurationUpdate(Map<String, Object> configurationParameter
             super.handleConfigurationUpdate(configurationParameters);
         }
     }
+
+    /**
+     * Remove previously statically created vane channel if the device does not support it. Or log a warning if it does
+     * support a vane and the respective channel is missing.
+     *
+     * @param bridgeHandler the calling bridge handler.
+     * @throws IllegalStateException if something went wrong.
+     */
+    public void updateDynamicChannels(VeluxBridgeHandler bridgeHandler) throws IllegalStateException {
+        // roller shutters are the only things with a previously statically created vane channel
+        if (!VeluxBindingConstants.THING_TYPE_VELUX_ROLLERSHUTTER.equals(thing.getThingTypeUID())) {
+            return;
+        }
+
+        String id = VeluxBindingConstants.CHANNEL_VANE_POSITION;
+        ChannelUID uid = new ChannelUID(thing.getUID(), id);
+        Thing2VeluxActuator actuator = new Thing2VeluxActuator(bridgeHandler, uid);
+        VeluxProduct product = bridgeHandler.existingProducts().get(actuator.getProductBridgeIndex());
+
+        if (product.equals(VeluxProduct.UNKNOWN)) {
+            throw new IllegalStateException("initializeVanePositionChannel(): Product unknown in the bridge");
+        }
+
+        Channel channel = thing.getChannel(id);
+        boolean required = product.supportsVanePosition();
+
+        if (!required && channel != null) {
+            logger.debug("Removing unsupported channel for {}: {}", thing.getUID(), id);
+            scheduler.submit(() -> updateThing(editThing().withoutChannels(channel).build()));
+        } else if (required && channel == null) {
+            logger.warn("Thing {} does not have a '{}' channel => please reinitialize it", thing.getUID(), id);
```

Comment:
Is it really "reinitialize" or rather "recreate" ?

If I correctly understand, there is no dynamic channel created by the binding. This is more a static channel which is removed if not expected. Is it correct ?

Reply:
> This is more a static channel which is removed if not expected. Is it correct ?

Correct. So indeed 'recreate' is better.

Reply:
Done.